src/users/users.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { IUser } from './interfaces/user.interface';
+
+
+@Injectable()
+export class UsersService {
+    private users: IUser[] = [];
+    private idCounter = 1;
+
+    create(createUserDto: CreateUserDto) {
```

Comment:
better to use interfaces in methods
we need to validate data only at entry point
in our case it's controller

Reply:
So please create interfaces)